### ISSUE TITLE
RFE: Please document backup best practices for journals

### ISSUE THREAD

**Issue by pabl0** [systemd/systemd#6673]
I've not been able to find any documentation on how to backup system journal files using traditional backup software without snapshots. I presume the active .journal file behaves like a live database, so a simple copy of the file might not be consistent after a restore? Can and should this be mitigated by rotating the journal just before backup starts and excluding the active .journal, or are there better strategies?

Traditional file-level backups have lots of problems but they are still very much in use, so it would be nice to have some guidance on how to properly handle systemd journals so that you don't end up with corrupted logs after a system restore. So please consider adding a section about backup to systemd-journald manual page.

**Comment by poettering** [comment]
Makes sense to document this.

It should be safe to just copy them. "journalctl" these days should be pretty OK to deal with partially inconsistent journal files. That said, you could run "journalctl --sync" first, so that everything queued is written to disk first, and the files are put in a guaranteed clean state. That said, given that your backup tool would run in parallel with journald then later writing new stuff to disk this only improves things a bit, things will still be racy...

**Comment by g0tar** [comment]
This is related, so I would like to share a solution I use for log retention, while preventing running out of disk space /etc/logrotate.d/journal:

```
/var/log/journal/*/user-*@*.journal
/var/log/journal/*/system@*.journal {
        olddir archive
        createolddir 750 root logs
        daily
        missingok
        nocreate
}

```

The problem with journal is that it cannot split journal files and apply different policies. E.g. keep my web server access logs for 12 months (legal retention), while dropping debug messages as soon as free space is needed. Since I don't wan't to go the old syslog-way, I choose to keep all of the journal files, as they compress very well (with xz I got c.a. 2% of the initial size). Logrotate catches the journals that were rotated by journald itself, so there is a window when the logs might be lost if running low disk space, but it is still better than loosing all the self-rotated logs everytime.
